viewing_party/party.py

```diff
@@ -0,0 +1,188 @@
+    return most_frequent
+
+# wave 3
+def get_unique_watched(user_data):
+    friend_movie_list = []
+    user_movie_list = []
+    unique_list = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movie_list.append(movie["title"])
+
+    for movie in user_data["watched"]:
+        user_movie_list.append(movie["title"])
+
+    unique_watched = set(user_movie_list) - set(friend_movie_list)
+    
+    for movie in unique_watched:
+        unique_list.append({"title": movie})
+    
+    return unique_list
+
+def get_friends_unique_watched(user_data):
+    friend_movie_list = []
+    user_movie_list = []
+    unique_list = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movie_list.append(movie["title"])
+    
+    for movie in user_data["watched"]:
+        user_movie_list.append(movie["title"])
+
+    unique_watched = set(friend_movie_list) - set(user_movie_list)
+    
+    for movie in unique_watched:
+        unique_list.append({"title": movie})
+    
+    return unique_list
+
+# wave 4
+def get_available_recs(user_data):
+    possible_recommendations = []
+    host_list = []
+    user_movie_list = []
+    final_list = []
+    friend_host_list = []
+
+    for movie in user_data["watched"]:
+        user_movie_list.append(movie["title"])
+
+    for host in user_data["subscriptions"]:
+        host_list.append(host)
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["host"] in host_list:
+                possible_recommendations.append(movie["title"])
+                friend_host_list.append(movie["host"])
+
+    recommendations = set(possible_recommendations) - set(user_movie_list)
+
+    if recommendations == False:
+        return None
+
+    for item in recommendations:
+        i = possible_recommendations.index(item)
+        final_list.append({"title": possible_recommendations[i], "host": friend_host_list[i]})
+    
+    return final_list
+
+# wave 5
+def get_new_rec_by_genre(user_data):
+    if user_data["watched"] == False:
+        return None
+
+    most_watched_genre = get_most_watched_genre(user_data)
+    possible_recommendations = []
+    user_movie_list = []
+    final_list = []
+
+    for movie in user_data["watched"]:
+        user_movie_list.append(movie["title"])
+    
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["genre"] == most_watched_genre:
+                possible_recommendations.append(movie["title"])
+
+    recommendations = set(possible_recommendations) - set(user_movie_list)
+    
+    if recommendations == False:
+        return None
+    
+    for item in recommendations:
+        i = possible_recommendations.index(item)
+        final_list.append({"title": possible_recommendations[i], "genre": most_watched_genre})
+    
+    return final_list
+    
+def get_rec_from_favorites(user_data):
```

Comment:
This function is very similar to the previous two, so my comments about rebuilding dictionaries applies here as well. We can think of this function as starting with the collection of movies the user has watched, but the friends have not (`get_unique_watched`) and then filtering this list down to those that the user has in their favorites list. This could even work with your wave 3 unchanged, since the movie contents here match the format from the previous wave, but I would still recommend thinking about how to approach these problems without building new dictionaries.